oracle/pkg/updater/updater.go

```diff
@@ -460,18 +471,66 @@ func (u *Updater) getL1Txns(ctx context.Context, blockNum uint64) (map[string]in
 
 	u.metrics.BlockTxnCacheMisses.Inc()
 
-	blk, err := u.l1Client.BlockByNumber(ctx, big.NewInt(0).SetUint64(blockNum))
+	block, err := u.l1Client.BlockByNumber(ctx, big.NewInt(0).SetUint64(blockNum))
 	if err != nil {
 		return nil, fmt.Errorf("failed to get block by number: %w", err)
 	}
 
-	txnsInBlock := make(map[string]int)
-	for posInBlock, tx := range blk.Transactions() {
-		txnsInBlock[strings.TrimPrefix(tx.Hash().Hex(), "0x")] = posInBlock
+	var txnReceipts sync.Map
+	eg, ctx := errgroup.WithContext(ctx)
+
+	txnsArray := make([]common.Hash, len(block.Transactions()))
+	for i, tx := range block.Transactions() {
+		txnsArray[i] = tx.Hash()
 	}
-	_ = u.l1BlockCache.Add(blockNum, txnsInBlock)
+	const bucketSize = 50 // Arbitrary number for bucket size
+
+	numBuckets := (len(txnsArray) + bucketSize - 1) / bucketSize // Calculate the number of buckets needed, rounding up
+	buckets := make([][]common.Hash, numBuckets)
+	for i := 0; i < numBuckets; i++ {
+		start := i * bucketSize
+		end := start + bucketSize
+		if end > len(txnsArray) {
+			end = len(txnsArray)
+		}
+		buckets[i] = txnsArray[start:end]
+	}
+
+	for _, bucket := range buckets {
+		bucket := bucket // closure for each errorgroup
```

Comment:
Ah so the closure is implicity,, because it's not overwriting the same bucket variable but decalring new ones on each iteration. Somewhat confusing behaviour before hand. Glad they updated this. Thanks for pointing this out! :)